File: app/src/main/java/dev/ragnarok/filegallery/api/HttpLogger.java
package dev.ragnarok.filegallery.api;

import dev.ragnarok.filegallery.BuildConfig;
import okhttp3.logging.HttpLoggingInterceptor;


public class HttpLogger {

    public static final HttpLoggingInterceptor DEFAULT_LOGGING_INTERCEPTOR = new HttpLoggingInterceptor();

    static {
        if (BuildConfig.DEBUG) {
            DEFAULT_LOGGING_INTERCEPTOR.setLevel(HttpLoggingInterceptor.Level.BODY);
        } else {
            DEFAULT_LOGGING_INTERCEPTOR.setLevel(HttpLoggingInterceptor.Level.NONE);
        }
    }
}

File: app/src/main/java/dev/ragnarok/filegallery/api/HttpLogger.kt
package dev.ragnarok.filegallery.api

import dev.ragnarok.filegallery.Constants
import okhttp3.logging.HttpLoggingInterceptor

object HttpLogger {
    val DEFAULT_LOGGING_INTERCEPTOR = HttpLoggingInterceptor()

    init {
        if (Constants.IS_DEBUG) {
            DEFAULT_LOGGING_INTERCEPTOR.setLevel(HttpLoggingInterceptor.Level.BODY)
        } else {
            DEFAULT_LOGGING_INTERCEPTOR.setLevel(HttpLoggingInterceptor.Level.NONE)
        }
    }
}